src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -61,6 +61,14 @@ public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageConte
         }
 
         /// <summary>
+        /// Creates a <see cref="IIncomingPhysicalMessageContext"/> based on the current context.
+        /// </summary>
+        public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageContext(this StageConnector<ITransportReceiveContext, IIncomingPhysicalMessageContext> stageConnector, IncomingMessage incomingMessage, ITransportReceiveContext sourceContext)
```

Comment:
Because there is absolutely no gain in replacing it. When using a satellite, you already have full control over the pipeline (which in this case consists of this single connector). That connector does not hide anything from the user.